File: python/counting_bits.py
import argparse
from typing import List


class Solution:
    def count_bits(self, n: int) -> List[int]:
        ans = []
        for i in range(n+1):
            bits = f"{i:b}"
            ans.append(bits.count("1"))
        return ans


def parse_arg():
    parser = argparse.ArgumentParser()
    parser.add_argument("--num", type=int)
    args = parser.parse_args()
    return args.num


if __name__ == "__main__":
    num = parse_arg()
    s = Solution()
    print(s.count_bits(num))

File: go/counting_bits.go
package main

import (
	"flag"
	"fmt"
)

func countBits(n int) []int {
    var ans []int
    for i := 0; i <= n; i++ {
        bits := fmt.Sprintf("%b", i)
        ones := 0
        for _, c := range bits {
            if string(c) == "1" {
                ones++
            }
        }
        fmt.Printf("Result for %d (%b) >> %d\n", i, i, ones)
        ans = append(ans, ones)
    }
    return ans
}

func countBitsBetter(n int) []int {
    ans := make([]int, n+1)
    for i:=0; i<=n ; i++ {
        k := i
        for k != 0 {
            ans[i]++
            k &= k -1
        }
    }
    return ans
}

func main() {
	num := flag.Int("num", 0, "Defines the intput value")
	flag.Parse()
    fmt.Println(countBitsBetter(*num))
}
